gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -250,6 +252,26 @@ public SubscriptionEntity create(NewSubscriptionEntity newSubscriptionEntity, St
                 throw new ApplicationArchivedException(applicationEntity.getName());
             }
 
+            if (applicationEntity.getApiKeyMode() == SHARED && planEntity.getSecurity().equals(API_KEY)) {
+                SubscriptionCriteria criteria = new SubscriptionCriteria.Builder()
```

Comment:
Subscriptions are already read from repository at line 275.

I wonder if we could use the already exiting `onlyValidSubs` do do this check, for example like it's done for JWT at line 300.
That would avoid an additional repository call.
WDYT ?